framework/components/dockercompose/billing_platform_service/billing_platform_service.go

```diff
@@ -0,0 +1,248 @@
+		Msgf("Starting Billing Platform Service stack with identifier %s",
+			framework.DefaultTCName(DEFAULT_STACK_NAME))
+
+	cFilePath, fileErr := composeFilePath(in.ComposeFile)
+	if fileErr != nil {
+		return nil, errors.Wrap(fileErr, "failed to get compose file path")
+	}
+
+	stack, stackErr := compose.NewDockerComposeWith(
+		compose.WithStackFiles(cFilePath),
+		compose.StackIdentifier(identifier),
+	)
+	if stackErr != nil {
+		return nil, errors.Wrap(stackErr, "failed to create compose stack for Billing Platform Service")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+
+	upErr := stack.Up(ctx)
+
+	if upErr != nil {
+		return nil, errors.Wrap(upErr, "failed to start stack for Billing Platform Service")
+	}
+
+	stack.WaitForService(DEFAULT_BILLING_PLATFORM_SERVICE_SERVICE_NAME,
+		wait.ForAll(
+			wait.ForLog("GRPC server is live").WithPollInterval(200*time.Millisecond),
+			wait.ForListeningPort(DEFAULT_BILLING_PLATFORM_SERVICE_GRPC_PORT),
+		).WithDeadline(1*time.Minute),
+	)
+
+	billingContainer, billingErr := stack.ServiceContainer(ctx, DEFAULT_BILLING_PLATFORM_SERVICE_SERVICE_NAME)
+	if billingErr != nil {
+		return nil, errors.Wrap(billingErr, "failed to get billing-platform-service container")
+	}
+
+	cli, cliErr := client.NewClientWithOpts(
+		client.FromEnv,
+		client.WithAPIVersionNegotiation(),
+	)
+	if cliErr != nil {
+		return nil, errors.Wrap(cliErr, "failed to create docker client")
+	}
+	defer cli.Close()
+
+	// so let's try to connect to a Docker network a couple of times, there must be a race condition in Docker
+	// and even when network sandbox has been created and container is running, this call can still fail
+	// retrying is simpler than trying to figure out how to correctly wait for the network sandbox to be ready
+	networks := []string{framework.DefaultNetworkName}
+	networks = append(networks, in.ExtraDockerNetworks...)
+
+	for _, networkName := range networks {
+		framework.L.Debug().Msgf("Connecting billing-platform-service to %s network", networkName)
+		connectContex, connectCancel := context.WithTimeout(ctx, 30*time.Second)
+		defer connectCancel()
+		if connectErr := connectNetwork(connectContex, 30*time.Second, cli, billingContainer.ID, networkName, identifier); connectErr != nil {
+			return nil, errors.Wrapf(connectErr, "failed to connect billing-platform-service to %s network", networkName)
+		}
+		// verify that the container is connected to framework's network
+		inspected, inspectErr := cli.ContainerInspect(ctx, billingContainer.ID)
+		if inspectErr != nil {
+			return nil, errors.Wrapf(inspectErr, "failed to inspect container %s", billingContainer.ID)
+		}
+
+		_, ok := inspected.NetworkSettings.Networks[networkName]
+		if !ok {
+			return nil, fmt.Errorf("container %s is NOT on network %s", billingContainer.ID, networkName)
+		}
+
+		framework.L.Debug().Msgf("Container %s is connected to network %s", billingContainer.ID, networkName)
+	}
```

Comment:
this probably could be extracted, defined as a function and used here and in chip ingress